pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,879 @@
+        Read the setting of RAMP.
+        """,
+        validator=strict_discrete_set,
+        values={i/10 for i in range(1000)} | {i for i in range(100, 10000)}
+        )
+
+    dynamic_temperature_setpoint = Instrument.measurement(
+        "SETD?",
+        """Read the dynamic temperature setpoint.
+        NOTE: This value changes during a temperature ramp to reflect the
+        instantaneous value at the time the query is executed.
+        """
+        )
+
+    load_setup_file = Instrument.setting(
+        "SFIL %g",
+        """loads setup file SFIL n where n is 1 - 12.
+        """,
+        validator=strict_range,
+        values=[1, 12]
+        )
+
+    enable_temperature_event_status = Instrument.control(
+        "TESE?", "TESE %g",
+        """ Set the temperature event status enable (mask) register.
+        TESE nnn -- where nnn is 0 – 255
+        
+        bit 7 - not used
+        bit 6 - not used
+        bit 5 - stopped cycling ("stop on fail" signal was received)
+        bit 4 - end of all cycles
+        bit 3 - end of one cycle
+        bit 2 - end of test (test time has elapsed)
+        bit 1 - not at temperature
+        bit 0 - at temperature (soak time has elapsed) 
+
+        NOTE: See TESR? for the meaning of each bit in the mask.
+        Read the temperature event status enable (mask) register.
+        """,
+        validator=strict_range,
+        values=[0, 255]
+        )
+
+    temperature_event_status = Instrument.measurement(
+        "TESR?",
+        """ temperature event status register.
+        
+        bit 7 - not used
+        bit 6 - not used
+        bit 5 - stopped cycling ("stop on fail" signal was received)
+        bit 4 - end of all cycles
+        bit 3 - end of one cycle
+        bit 2 - end of test (test time has elapsed)
+        bit 1 - not at temperature
+        bit 0 - at temperature (soak time has elapsed) 
+        """,
+        validator=strict_range,
+        values=[0, 255]
+        )
+
+    air_temperature = Instrument.measurement(
+        "TMPA?",
+        """Read air temperature, in 0.1 °C increments.
+
+        NOTE: This query always returns the air temperature, whether in
+        air or DUT control modes.
+        """
+        )
+
+    dut_temperature = Instrument.measurement(
+        "TMPD?",
+        """Read DUT temperature, in 0.1 °C increments.
+        
+        NOTE: This query always returns the DUT temperature, whether in
+        air or DUT control modes.
+        """
+        )
+
+    mode = Instrument.measurement(
+        "WHAT?",
+        """Returns an integer indicating what the system is doing at the time the query is processed.
+        
+        5 = on Operator screen (manual mode)
+        6 = on Cycle screen (program mode)
+        """,
+        values={'manual' : 5, 
+                'program'  : 6},
+        map_values=True
+        )
+
+    def __init__(self, adapter, delay=0.02, **kwargs):
```

Comment:
you are not using the `delay` argument?

Reply:
you are right, it should be removed there.

Reply:
I removed it now